### ISSUE TITLE
Solid model is missing for an assembled part that was forced to triangle mesh at design time

### ISSUE THREAD

**Issue by blinkenlight** [solvespace/solvespace#1269]
### System information

- **SolveSpace version:** 3.1~d6e1b230
- **Operating system:** Windows 7 x64

### Expected behavior

Triangle-mesh assemblies should have a visible solid model

### Actual behavior

Assembled part that had beed restricted to triangle mesh at design time loads with no visible solid model as an assembly

### Additional information

Using the model attached in #1268 except with ticking "restrict to triangle mesh" on the cylinder group to avoid the naked edges issue, loading it as an assembly there is no solid model of it visible. Also ticking the "restrict to triangle mesh" option in the group created by loading this model as an assembly makes the solid model appear. This is, however, extremely counter-intuitive until one finally figures out this needs to be done to see the solid model.
![Solid Model Error 1a](https://user-images.githubusercontent.com/2522619/178161477-88c44fa0-4e2a-40c9-8114-c98bd005593b.jpg)
![Solid Model Error 1b](https://user-images.githubusercontent.com/2522619/178161480-bd90dc81-b5f6-4baf-94c4-574f45cbaa7a.jpg)

**Comment by phkahler** [comment]
@blinkenlight I'm speculating at the pub. If you force the model to triangle mesh in the first extrusion instead of the last one, does it link properly?

**Comment by blinkenlight** [comment]
Nope. No matter which group (extrusion or sketch) I use to restrict to triangle mesh, the result is the same - no solid model as an assembly unless I also restrict the assembly itself... (also: cheers! ;))

**Comment by ruevs** [comment]
https://solvespace.com/forum.pl?action=viewthread&parent=4791

**Comment by phkahler** [comment]
I've made a local branch to fix this. The best way to fix it seems to be to modify the function Group::IsTriangleMeshAssembly()  which just checks if we linked an STL file. A better test is if the linked group has a triangle mesh but no NURBS shell, as that should cover every case including STL and OBJ if we ever do it, while also flagging slvs files that end with a mesh but not shell.

I'm not sure how to check that the group has a mesh or shell though. Poking around the code now...
